modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/DataAccessMappingFeatureIterator.java

```diff
@@ -760,17 +760,6 @@ protected Attribute setAttributeValue(
             Collection<Object> cobjs = (Collection<Object>) values;
             for (Object singleVal : cobjs) {
                 ArrayList<Object> valueList = new ArrayList<>();
-                // copy client properties from input features if they're complex features
-                // wrapped in app-schema data access
-                if (singleVal instanceof Attribute) {
-                    // copy client properties from input features if they're complex features
-                    // wrapped in app-schema data access
-                    final Map<Name, Expression> valueProperties =
-                            getClientProperties((Attribute) singleVal);
-                    if (!valueProperties.isEmpty()) {
-                        clientPropsMappings.putAll(valueProperties);
-                    }
-                }
```

Comment:
Very interesting... so you remove this bit of code, and no test is failing? What was it for? The git history does not say, so I went back into SVN and found this commit:

```
------------------------------------------------------------------------
r34494 | ang05a | 2009-11-25 13:15:49 +0100(mer, 25 nov 2009) | 1 line

Web service data access support: copy underlying client properties to app schema data access (take 2)
------------------------------------------------------------------------
```

I could not find a ticket for it. Does it ring any bell to you?
There was a way to integrate app-schema with a web-service, but I believe we removed it in the meantime.